FWCore/Framework/src/Schedule.cc

```diff
@@ -232,6 +232,12 @@ namespace edm {
 
       std::map<BranchKey, BranchKey> aliasKeys;  // Used to search for duplicates or clashes.
 
+      // Auxiliary search structure to support wildcard for friendlyClassName
+      std::multimap<std::string, BranchKey> moduleLabelToBranches;
+      for (auto const& prod : preg.productList()) {
+        moduleLabelToBranches.emplace(prod.first.moduleLabel(), prod.first);
```

Comment:
Would it be simpler to require the current process name before inserting into this map? Then you wouldn't need to check that in two places below. Also less entries in the map would lead to a performance gain (probably insignificant).

Reply:
Thanks for the suggestion, I agree, done.